File: trunk/C#/Code/SPTemplate.UI/WebControl/ErrorCaptureControl.cs
using System;
using System.Linq;
using System.Text;
using System.Web.UI.WebControls;
using System.Collections;
using Microsoft.SharePoint;
using Microsoft.SharePoint.WebControls;
using System.Web.UI;
using System.Web;
using $customUsing$.Common;
using $customUsing$.Exceptions;

namespace $safeprojectname$.WebControls
{
    public class ErrorCaptureControl : WebControl
    {
       

        #region Render Message
        protected override void OnPreRender(EventArgs e)
        {
        }

        protected override void Render(HtmlTextWriter writer)
        {
            String sMessage = Convert.ToString(HttpContext.Current.Cache[CommonFunctions.GetErrorCodeKey(HttpContext.Current, SPContext.Current.Web)]);
            if (!String.IsNullOrEmpty(sMessage))
            {
                DisplayMessage(sMessage);
                ClearMessageFromUI();

                if (sMessage.Contains(Constants.UNIQUE_KEY_CODE_SUCESS))
                {
                    //unique will determine that the message needs to be displayed on second post back; case with long operations
                    sMessage = sMessage.Replace(Constants.UNIQUE_KEY_CODE_SUCESS, String.Empty);
                    SetSuccessMessageforUI(sMessage);
                }
                else if (sMessage.Contains(Constants.UNIQUE_KEY_CODE_FAILURE))
                {
                    sMessage = sMessage.Replace(Constants.UNIQUE_KEY_CODE_FAILURE, String.Empty);
                    SetErrorMessageforUI(sMessage);
                }
            }
        }
        #endregion

       #region Set Message in UI
        public static void ClearMessageFromUI()
        {
            if (HttpContext.Current != null)
            {
                CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, String.Empty);
            }
        }

        public static void ClearMessageFromUI(SPWeb web)
        {
            if (HttpContext.Current != null)
            {
                CustomExceptions.SetMessageVariable(HttpContext.Current, web, String.Empty);
            }
        }

        public static void SetErrorMessageforUI(String sErrorMessage)
        {
            if (HttpContext.Current != null)
            {
                if (!String.IsNullOrEmpty(sErrorMessage))
                {
                    String sMessage = Constants.ERROR_MESSAGE_SYMBOL + sErrorMessage;
                    CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage);
                }
            }
        }

        public static void SetErrorMessageforUI(SPWeb web, String sErrorMessage)
        {
            if (HttpContext.Current != null)
            {
                if (!String.IsNullOrEmpty(sErrorMessage))
                {
                    String sMessage = Constants.ERROR_MESSAGE_SYMBOL + sErrorMessage;
                    CustomExceptions.SetMessageVariable(HttpContext.Current, web, sMessage);
                }
            }
        }

        public static void SetSuccessMessageforUI(String sErrorMessage)
        {
            if (HttpContext.Current != null)
            {
                if (!String.IsNullOrEmpty(sErrorMessage))
                {
                    String sMessage = Constants.SUCCESS_MESSAGE_SYMBOL + sErrorMessage;
                    CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage);
                }
            }
        }

        public static void SetSuccessMessageforUI(SPWeb web, String sErrorMessage)
        {
            if (HttpContext.Current != null)
            {
                if (!String.IsNullOrEmpty(sErrorMessage))
                {
                    String sMessage = Constants.SUCCESS_MESSAGE_SYMBOL + sErrorMessage;
                    CustomExceptions.SetMessageVariable(HttpContext.Current, web, sMessage);
                }
            }
        }

        public static void SetSuccessMessageforLongOperations(String sMessage)
        {
            if (HttpContext.Current != null)
            {
                CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage + Constants.UNIQUE_KEY_CODE_SUCESS);
            }
        }

        public static void SetErrorMessageforLongOperations(String sMessage)
        {
            if (HttpContext.Current != null)
            {
                CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage + Constants.UNIQUE_KEY_CODE_FAILURE);
            }
        }
	    #endregion

        #region Private Methods
 

        private void DisplayMessage(String strMessage)
        {
            if (!String.IsNullOrEmpty(strMessage) && strMessage.Length > 1)
            {
                String sSymbol = strMessage.Substring(0, 1);
                strMessage = strMessage.Substring(1);
                strMessage = strMessage.Replace("'", "&#39;");

                switch (sSymbol)
                {
                    case Constants.ERROR_MESSAGE_SYMBOL:
                        ShowFailureMessage(strMessage);
                        break;
                    case Constants.SUCCESS_MESSAGE_SYMBOL:
                        ShowSuccessMessage(strMessage);
                        break;
                }
            }
        }
      

        private void ShowFailureMessage(String strMessage)
        {
            String strFormatString = String.Empty;
            try
            {
                strMessage = strMessage.Replace("\\r", "\\ r");
                strFormatString = String.Format("javascript:ShowFailureStatusBarMessage('{0}','{1}');", "Error :", strMessage);
                RegisterScript(strFormatString);
            }
            catch (Exception ex)
            {

            }
        }

        private void ShowSuccessMessage(String strMessage)
        {
            String strFormatString = String.Empty;
            try
            {
                strFormatString = String.Format("javascript:ShowSuccessStatusBarMessage('{0}','{1}');", "Note :", strMessage);
                RegisterScript(strFormatString);
            }
            catch (Exception ex)
            {

            }
        }

        private void RegisterScript(String strFormatString)
        {
            String strJavaScript = String.Empty;
            strJavaScript = "<script type=\"text/javascript\">ExecuteOrDelayUntilScriptLoaded(Initialize, \"sp.js\");function Initialize(){ " + strFormatString + " }</script>";
            ScriptManager.RegisterStartupScript(this.Page, this.Page.GetType(), "opennotification", strJavaScript, false);
        }

        #endregion
    }
}

File: trunk/VB/Code/SPTemplate.UI/WebControl/ErrorCaptureControl.vb
Imports System.Linq
Imports System.Text
Imports System.Web.UI.WebControls
Imports System.Collections
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.WebControls
Imports System.Web.UI
Imports System.Web
Imports $customUsing$.Common
Imports $customUsing$.Exceptions

Namespace WebControls
    Public Class ErrorCaptureControl
        Inherits WebControl


#Region "Render Message"
        Protected Overrides Sub OnPreRender(ByVal e As EventArgs)
        End Sub

        Protected Overrides Sub Render(ByVal writer As HtmlTextWriter)
            Dim sMessage As [String] = Convert.ToString(HttpContext.Current.Cache(CommonFunctions.GetErrorCodeKey(HttpContext.Current, SPContext.Current.Web)))
            If Not [String].IsNullOrEmpty(sMessage) Then
                DisplayMessage(sMessage)
                ClearMessageFromUI()

                If sMessage.Contains(Constants.UNIQUE_KEY_CODE_SUCESS) Then
                    'unique will determine that the message needs to be displayed on second post back; case with long operations
                    sMessage = sMessage.Replace(Constants.UNIQUE_KEY_CODE_SUCESS, [String].Empty)
                    SetSuccessMessageforUI(sMessage)
                ElseIf sMessage.Contains(Constants.UNIQUE_KEY_CODE_FAILURE) Then
                    sMessage = sMessage.Replace(Constants.UNIQUE_KEY_CODE_FAILURE, [String].Empty)
                    SetErrorMessageforUI(sMessage)
                End If
            End If
        End Sub
#End Region

#Region "Set Message in UI"
        Public Shared Sub ClearMessageFromUI()
            If HttpContext.Current IsNot Nothing Then
                CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, [String].Empty)
            End If
        End Sub

        Public Shared Sub ClearMessageFromUI(ByVal web As SPWeb)
            If HttpContext.Current IsNot Nothing Then
                CustomExceptions.SetMessageVariable(HttpContext.Current, web, [String].Empty)
            End If
        End Sub

        Public Shared Sub SetErrorMessageforUI(ByVal sErrorMessage As [String])
            If HttpContext.Current IsNot Nothing Then
                If Not [String].IsNullOrEmpty(sErrorMessage) Then
                    Dim sMessage As [String] = Constants.ERROR_MESSAGE_SYMBOL + sErrorMessage
                    CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage)
                End If
            End If
        End Sub

        Public Shared Sub SetErrorMessageforUI(ByVal web As SPWeb, ByVal sErrorMessage As [String])
            If HttpContext.Current IsNot Nothing Then
                If Not [String].IsNullOrEmpty(sErrorMessage) Then
                    Dim sMessage As [String] = Constants.ERROR_MESSAGE_SYMBOL + sErrorMessage
                    CustomExceptions.SetMessageVariable(HttpContext.Current, web, sMessage)
                End If
            End If
        End Sub

        Public Shared Sub SetSuccessMessageforUI(ByVal sErrorMessage As [String])
            If HttpContext.Current IsNot Nothing Then
                If Not [String].IsNullOrEmpty(sErrorMessage) Then
                    Dim sMessage As [String] = Constants.SUCCESS_MESSAGE_SYMBOL + sErrorMessage
                    CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage)
                End If
            End If
        End Sub

        Public Shared Sub SetSuccessMessageforUI(ByVal web As SPWeb, ByVal sErrorMessage As [String])
            If HttpContext.Current IsNot Nothing Then
                If Not [String].IsNullOrEmpty(sErrorMessage) Then
                    Dim sMessage As [String] = Constants.SUCCESS_MESSAGE_SYMBOL + sErrorMessage
                    CustomExceptions.SetMessageVariable(HttpContext.Current, web, sMessage)
                End If
            End If
        End Sub

        Public Shared Sub SetSuccessMessageforLongOperations(ByVal sMessage As [String])
            If HttpContext.Current IsNot Nothing Then
                CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage + Constants.UNIQUE_KEY_CODE_SUCESS)
            End If
        End Sub

        Public Shared Sub SetErrorMessageforLongOperations(ByVal sMessage As [String])
            If HttpContext.Current IsNot Nothing Then
                CustomExceptions.SetMessageVariable(HttpContext.Current, SPContext.Current.Web, sMessage + Constants.UNIQUE_KEY_CODE_FAILURE)
            End If
        End Sub
#End Region

#Region "Private Methods"


        Private Sub DisplayMessage(ByVal strMessage As [String])
            If Not [String].IsNullOrEmpty(strMessage) AndAlso strMessage.Length > 1 Then
                Dim sSymbol As [String] = strMessage.Substring(0, 1)
                strMessage = strMessage.Substring(1)
                strMessage = strMessage.Replace("'", "&#39;")

                Select Case sSymbol
                    Case Constants.ERROR_MESSAGE_SYMBOL
                        ShowFailureMessage(strMessage)
                        Exit Select
                    Case Constants.SUCCESS_MESSAGE_SYMBOL
                        ShowSuccessMessage(strMessage)
                        Exit Select
                End Select
            End If
        End Sub


        Private Sub ShowFailureMessage(ByVal strMessage As [String])
            Dim strFormatString As [String] = [String].Empty
            Try
                strMessage = strMessage.Replace("\r", "\ r")
                strFormatString = [String].Format("javascript:ShowFailureStatusBarMessage('{0}','{1}');", "Error :", strMessage)
                RegisterScript(strFormatString)

            Catch ex As Exception
            End Try
        End Sub

        Private Sub ShowSuccessMessage(ByVal strMessage As [String])
            Dim strFormatString As [String] = [String].Empty
            Try
                strFormatString = [String].Format("javascript:ShowSuccessStatusBarMessage('{0}','{1}');", "Note :", strMessage)
                RegisterScript(strFormatString)

            Catch ex As Exception
            End Try
        End Sub

        Private Sub RegisterScript(ByVal strFormatString As [String])
            Dim strJavaScript As [String] = [String].Empty
            strJavaScript = "<script type=""text/javascript"">ExecuteOrDelayUntilScriptLoaded(Initialize, ""sp.js"");function Initialize(){ " & Convert.ToString(strFormatString) & " }</script>"
            ScriptManager.RegisterStartupScript(Me.Page, Me.Page.[GetType](), "opennotification", strJavaScript, False)
        End Sub

#End Region
    End Class
End Namespace
